db/migrate/20230531120502_create_transactions.rb

```diff
@@ -0,0 +1,14 @@
+class CreateTransactions < ActiveRecord::Migration[7.0]
+  def change
+    create_table :transactions do |t|
+      t.references :recipient, null: true, foreign_key: { to_table: :users }
+      t.references :payer, null: true, foreign_key: { to_table: :users }
+      t.string :recipient_name
+      t.string :payer_name
+      t.monetize :amount
+      t.string :communication
+
+      t.timestamps
+    end
+  end
+end
```

Comment:
could you please explain me why do you allow the `recipient` & `payer` columns to have NULL values ?

Reply:
Because if I user deletes his account I still need the transactions to persist otherwise Chloe's accountability will become wrong.
That's why I store the users' full name too, so that even if the user destroys his account the Transaction still has an indicator of who was the payer and the recipient

Reply:
Ok, that is clear and logical. Thanks ;)